### DISCUSSION TITLE
Fork Uniswap for a new Ethereum chain gives MetaMask - RPC Error

### DISCUSSION THREAD

**Discussion by Ben-Haslam** [Uniswap/interface#1367] [Q&A]
**Bug Description**

I have forked the Uniswap interface for a private ethereum network, by adding another chain Id that the interface can use and modifying the right addresses. (this means rebuilding the sdk too and importing the `dist` into the `uniswap-interface` node modules. )

Upon loading the development build of the interface and connecting to a metamask wallet, the interface initially appears successful, displaying the balance of ETH and tokens, but then after a few seconds loses connection. If I look at the console in browser, I see that when it loses connection, I get the following error:

![image](https://user-images.githubusercontent.com/59572435/112996878-7e7fe480-9164-11eb-9985-45ea2f2359c1.png)

If anybody has seen this problem before and might know how to fix it I would be extremely thankful!

**Steps to Reproduce**
1. Go to the repo for the forked interface [here](https://github.com/clearmatics/uniswap-interface/blob/master/README.md), or make your own fork for a new Ethereum network
2. Run the script `script.sh`

**Expected Behavior**
The interface should work with the Autontiy Bakerloo network (or another network it has been forked for) without losing connection and giving this error.

**Additional Context**

I have also tried using a repo made by @tpmccallum to do the same thing [here](https://github.com/second-state/how_to_deploy_uniswap) which led to the same outcome.

I believe I am not alone in having this error when attempting to fork the interface for another network, as these issues show the same problem:

https://github.com/makerdao/multicall/issues/14
https://github.com/second-state/how_to_deploy_uniswap/issues/4

**Comment by 0xchain1** [comment]
Did you ever figure out the issue? I am facing the same problem
